src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php

```diff
@@ -462,8 +479,15 @@ private function createNewInvoice(Order $order, Cart $cart, $isFreeShipping)
         $taxCalculator = $carrier->getTaxCalculator($invoice_address);
 
         // @todo: use https://github.com/PrestaShop/decimal to compute prices and taxes
-        $invoice->total_paid_tax_excl = Tools::ps_round((float) $cart->getOrderTotal(false, $totalMethod), 2);
-        $invoice->total_paid_tax_incl = Tools::ps_round((float) $cart->getOrderTotal(true, $totalMethod), 2);
+        $precision = $this->getPrecisionFromCart($cart);
+        $invoice->total_paid_tax_excl = Tools::ps_round(
```

Comment:
Same here i you create a new invoice with a cart containing ALL the products the total won't be correct

Reply:
Changed the behavior.